src/main/java/lotto/domain/Lotto.java

```diff
@@ -18,22 +21,25 @@ public Lotto() {
     }
 
     public Lotto(List<Integer> numbers) {
-        List<LottoNumber> lottoNumbers = numbers.stream().map(LottoNumber::new).collect(Collectors.toList());
-        checkArgumentValidation(lottoNumbers);
-        this.numbers = new TreeSet<>(lottoNumbers);
+        this(numbers.stream().map(LottoNumber::new).collect(Collectors.toCollection(TreeSet::new)));
     }
 
-    private void checkArgumentValidation(List<LottoNumber> numbers) {
+    public Lotto(Set<LottoNumber> numbers) {
+        this.numbers = checkArgumentValidation(numbers);
+    }
+
+    private Set<LottoNumber> checkArgumentValidation(Set<LottoNumber> numbers) {
         if (numbers.size() != SIZE) {
             throw new IllegalArgumentException("Lotto의 List<Integer> size는 항상 6입니다.");
         }
+        return numbers;
     }
 
     public Set<LottoNumber> getNumbers() {
         return this.numbers;
     }
```

Comment:
컬렉션을 그대로 반환하면 변조를 할 수 있어 일급 컬렉션의 의미가 없어지는 것 같아요.
내부 요소를 수정하지 못하는 컬렉션을 반환하면 어떨까요?
```suggestion
    public Set<LottoNumber> getNumbers() {
        return Collections.unmodifiableList(numbers);
    }
```